mailpoet/lib/Subscription/Registration.php

```diff
@@ -63,19 +63,39 @@ public function extendForm() {
     print $form;
   }
 
-  public function onMultiSiteRegister($result) {
-    if (empty($result['errors']->errors)) {
-      if (
-        isset($_POST['mailpoet']['subscribe_on_register'])
-        && (bool)$_POST['mailpoet']['subscribe_on_register'] === true
-      ) {
-        $this->subscribeNewUser(
-          $result['user_name'],
-          $result['user_email']
-        );
-      }
+  public function checkMultsiteRegister() {
```

Comment:
To understand this, I have a dumb question at first. Why don't we just register [the tracking part](https://github.com/mailpoet/mailpoet/pull/4064/files#diff-13238361a5ac4b7b8247fcc268aa39e82c46483b5b464abc224d5400dc0dc4deL113) later? Like `$this->wp->addAction(...)` instead of direct `$this->subscriberHandler->identifyByEmail($email);`?

(Or maybe even [the subscriber handler](https://github.com/mailpoet/mailpoet/blob/fc1f3e6dc265f5617c44b49ac80fb149f151c43a/mailpoet/lib/Statistics/Track/SubscriberHandler.php#L60) could do it?)